File: celladapter/src/main/java/io/erva/celladapter/select/mode/MultiSelectionManager.java
package io.erva.celladapter.select.mode;

import java.util.List;

public class MultiSelectionManager extends SelectionManager {

    @Override
    public void toggleSelection(Integer position) {
        super.toggleSelection(position);
        adapter.notifyItemChanged(position);
    }

    @Override
    public void setSelection(Integer position, boolean isSelected) {
        super.setSelection(position, isSelected);
        adapter.notifyItemChanged(position);
    }

    @Override
    public boolean isSelected(int position) {
        return getSelectedPositions().contains(position);
    }

    public boolean isAllSelected() {
        return getSelectedItemCount() == adapter.getItemCount();
    }

    public void setSelectedPositions(List<Integer> selectionPositions) {
        clearSelections(false);
        for (Integer position : selectionPositions) toggleSelection(position);
        adapter.notifyDataSetChanged();
    }

    public void setSelectionForAll(boolean isSelected) {
        for (int position = 0; position < adapter.getItemCount(); position++)
            setSelection(position, isSelected);
    }
}

File: celladapter-kotlin/src/main/kotlin/io/erva/celladapter/v7/select/mode/MultiSelectionManager.kt
package io.erva.celladapter.v7.select.mode

class MultiSelectionManager : SelectionManager() {

    override fun toggleSelection(position: Int) {
        super.toggleSelection(position)
        adapter.notifyItemChanged(position)
    }

    override fun setSelection(position: Int, isSelected: Boolean) {
        super.setSelection(position, isSelected)
        adapter.notifyItemChanged(position)
    }

    override fun isSelected(position: Int): Boolean {
        return getSelectedPositions().contains(position)
    }

    fun isAllSelected(): Boolean {
        return getSelectedItemCount() == adapter.itemCount
    }

    fun setSelectedPositions(selectionPositions: List<Int>) {
        clearSelections(false)
        for (position in selectionPositions) toggleSelection(position)
        adapter.notifyDataSetChanged()
    }

    fun setSelectionForAll(isSelected: Boolean) {
        for (position in 0 until adapter.itemCount)
            setSelection(position, isSelected)
    }
}